lib/spack/spack/environment.py

```diff
@@ -598,3 +599,28 @@ def inspect_path(root, inspections, exclude=None):
                 env.prepend_path(variable, expected)
 
     return env
+
+
+@contextlib.contextmanager
+def preserve_environment(*variables):
+    """Ensures that the value of the environment variables passed as
+    arguments is the same before entering to the context manager and after
+    exiting it.
```

Comment:
I think it would be worth adding that variables which are unset before the context manager will be explicitly unset after.